graylog2-web-interface/src/pages/LoginPage.jsx

```diff
@@ -33,9 +35,33 @@ const LoginPageStyles = createGlobalStyle`
   ${authStyles}
 `;
 
+const StyledButton = styled(Button)`
+  margin-top: 1em;
+  display: inline-block;
+  cursor: pointer;
+`;
+
 const LoginPage = () => {
   const [didValidateSession, setDidValidateSession] = useState(false);
   const [lastError, setLastError] = useState(undefined);
+  const [activeBackend, setActiveBackend] = useState();
+  const [isDetermined, setIsDetermined] = useState(false);
+  const [useFallback, setUseFallback] = useState(false);
+
+  const isCloud = AppConfig.isCloud();
+  const registeredLoginComponents = PluginStore.exports('loginProviderType');
+  const loginComponent = registeredLoginComponents.find((c) => c.type === activeBackend);
+  const hasCustomLogin = loginComponent && loginComponent.formComponent;
+
+  useEffect(() => {
```

Comment:
This works fine and there is nothing wrong with it. We could make it a bit simpler by using `react-query`. In this case it would look like this:

```
  const { data: activeBackend, isSuccess: isDetermined } = useQuery('activeBackendType', AuthenticationDomain.loadActiveBackendType);
```

I am mapping parts of the response to the names you used (`activeBackend`, `isDetermined`), but this is just for illustration and you can also use the more idiomatic name (`isSuccess`), plus you get more (`isError`/`error`/...).

As a plus, you could wrap this in a custom hook straight-away:
```
const useActiveBackendType = () => useQuery('activeBackendType', AuthenticationDomain.loadActiveBackendType);

[...]

const LoginPage = () => {
[...]
  const { data: activeBackend, isSuccess: isDetermined } = useActiveBackendType();
```

This could be reused in other places then as well.

Reply:
Thanks for the tip @dennisoelkers. Let me dig into this. Never used `react-query`, but it looks like how I'd do hooks that fetch data in my own projects.

Reply:
Updated with you suggestions @dennisoelkers